pkg/runtime-manager/server/cri/criserver.go

```diff
@@ -0,0 +1,129 @@
+	"github.com/koordinator-sh/koordinator/pkg/runtime-manager/dispatcher"
+	resource_executor "github.com/koordinator-sh/koordinator/pkg/runtime-manager/resource-executor"
+	"github.com/koordinator-sh/koordinator/pkg/runtime-manager/server/utils"
+	"google.golang.org/grpc"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"k8s.io/klog/v2"
+)
+
+type RuntimeManagerCriServer struct {
+	dispatcher    *dispatcher.RuntimeHookDispatcher
+	runtimeClient runtimeapi.RuntimeServiceClient
+}
+
+func NewRuntimeManagerCriServer(dispatcher *dispatcher.RuntimeHookDispatcher) *RuntimeManagerCriServer {
+	criInterceptor := &RuntimeManagerCriServer{
+		dispatcher: dispatcher,
+	}
+	return criInterceptor
+}
+
+func (ci *RuntimeManagerCriServer) Name() string {
+	return "RuntimeManagerCriServer"
+}
+
+func (ci *RuntimeManagerCriServer) Setup() error {
+	if err := ci.initBackendServer(utils.DefaultContainerdSocketPath); err != nil {
+		return err
+	}
+	lis, err := net.Listen("unix", utils.DefaultRuntimeManagerSocketPath)
+	if err != nil {
+		klog.Errorf("fail to create the lis %v", err)
+		return err
+	}
+	grpcServer := grpc.NewServer()
+	runtimeapi.RegisterRuntimeServiceServer(grpcServer, ci)
+	err = grpcServer.Serve(lis)
+	return err
+}
+
+func (ci *RuntimeManagerCriServer) getRuntimeHookInfo(serviceType RuntimeServiceType) (config.RuntimeRequestPath,
+	resource_executor.RuntimeResourceType) {
+	switch serviceType {
+	case RunPodSandbox:
+		return config.RunPodSandbox, resource_executor.RuntimePodResource
+	case CreateContainer:
+		// No Nook point in create container, but we need store the container info during container create
+		return config.NoneRuntimeHookPath, resource_executor.RuntimeContainerResource
+	case StartContainer:
+		return config.StartContainer, resource_executor.RuntimeContainerResource
+	case StopContainer:
+		return config.StopContainer, resource_executor.RuntimeContainerResource
+	case UpdateContainerResources:
+		return config.UpdateContainerResources, resource_executor.RuntimeContainerResource
+	}
+	return config.NoneRuntimeHookPath, resource_executor.RuntimeNoopResource
+}
+
+func (ci *RuntimeManagerCriServer) interceptRuntimeRequest(serviceType RuntimeServiceType,
+	ctx context.Context, request interface{}, handler grpc.UnaryHandler) (interface{}, error) {
+
+	runtimeHookPath, runtimeResourceType := ci.getRuntimeHookInfo(serviceType)
+	resourceExecutor := resource_executor.NewRuntimeResourceExecutor(runtimeResourceType)
+
+	if err := resourceExecutor.ParseRequest(request); err != nil {
+		klog.Errorf("fail to parse request %v %v", request, err)
+	}
+
+	// pre call hook server
+	if _, err := ci.dispatcher.Dispatch(ctx, runtimeHookPath, config.PreHook, resourceExecutor.GenerateHookRequest()); err != nil {
+		klog.Errorf("fail to call hook server %v", err)
+	}
+
+	// call the backend runtime engine
+	res, err := handler(ctx, request)
+	if err == nil {
+		klog.Infof("%v %v success", res, err)
```

Comment:
we should also consider logging the basic information of the request such as Pod ID, container ID etc.

Reply:
Or record requests and response through middleware

Reply:
introduce GetMetaInfo to resourceExecutor for podID/containerID show

Reply:
> introduce GetMetaInfo to resourceExecutor for podID/containerID show

introduce resourceExecutor.GetMetaInfo  for podID/containerID show

Reply:
> we should also consider logging the basic information of the request such as Pod ID, container ID etc.

introduce resourceExecutor.GetMetaInfo for podID/containerID show

Reply:
I think we need a middleware to track request/response that can help quickly locate and analyze problems.

Reply:
> I think we need a middleware to track request/response that can help quickly locate and analyze problems.

trace in a new issue #109